play/src/front/Components/VisitCard/VisitCard.svelte

```diff
@@ -1,19 +1,53 @@
 <script lang="ts">
     import { fly } from "svelte/transition";
     import { onMount } from "svelte";
-    import { requestVisitCardsStore } from "../../Stores/GameStore";
+    import { get } from "svelte/store";
+    import { requestVisitCardsStore, selectedChatIDRemotePlayerStore } from "../../Stores/GameStore";
     import { LL } from "../../../i18n/i18n-svelte";
+    import { chatVisibilityStore } from "../../Stores/ChatStore";
+    import { gameManager } from "../../Phaser/Game/GameManager";
+    import { navChat, selectedRoom } from "../../Chat/Stores/ChatStore";
+    import { ChatRoom } from "../../Chat/Connection/ChatConnection";
 
     export let visitCardUrl: string;
     let w = "500px";
     let h = "250px";
     let hidden = true;
     let cvIframe: HTMLIFrameElement;
 
+    const chatConnection = gameManager.getCurrentGameScene().chatConnection;
+    const selectPlayerChatID = get(selectedChatIDRemotePlayerStore);
+
     function closeCard() {
         requestVisitCardsStore.set(null);
     }
 
+    function canSendMessageTo(chatID: string): boolean {
+        let room: ChatRoom | undefined = chatConnection.getDirectRoomFor(chatID);
+
+        const isGuest = get(chatConnection.isGuest) ?? true;
+
+        if (!room && isGuest) return false;
+
+        return true;
+    }
+
+    async function sendMessage() {
```

Comment:
Rename it. This method is not about sending a message but opening the chat to the direct room. It could be confusing.